syntax = "proto2";


enum state  {
  AndhraPradesh = 0;
  ArunachalPradesh = 1;
  Assam = 2;
  Bihar = 3;
  Chhattisgarh = 4;
  Goa = 5;
  Gujarat = 6;
  Haryana = 7;
  HimachalPradesh = 8;
  JammuandKashmir = 9;
  Jharkhand = 10;
  Karnataka = 11;
  Kerala = 12;
  MadyaPradesh =13;
  Maharashtra= 14;
  Manipur= 15;
  Meghalaya = 16;
  Mizoram = 17;
  Nagaland = 18;
  Orissa = 19;
  Punjab = 20;
  Rajasthan = 21;
  Sikkim = 22;
  TamilNadu = 23;
  Telagana = 24;
  Tripura = 25;
  Uttaranchal = 26;
  UttarPradesh = 27;
  WestBengal = 28;
}

enum type{
  vegetable_shortt = 1;
  vegetable_longg = 2;
  fruits_shortt = 3;
  fruits_longg = 4;
  grainss = 5;
  pulsess = 6;

}
//Short means life span == 2 days max (most hard to deal with)
//Long means life span 2 weeks to 1 year (onion)
//Grains have a great life span so don't worry
enum vegetable_short {
  Tomato = 0;
  Lemons = 1;
  Cauliflower = 2;
  Spinach = 3;
}

enum vegetable_long {
  Potato = 0;
  Onion = 1;
  Garlic = 2;
  Cabbage = 3;
  Beets = 4;
}

enum fruits_short{
  Strawberries =0;
  Cherries = 1;
  Bananas =2 ;
  Pineapple = 3;
}

enum fruits_long{
  Apples = 0;
  Orange = 1;
  Pomegranate = 2;
}

enum grains{
  Wheat = 0;
  Rice = 1;
  Maize =2;
  Barley = 3;
}
// lasts for a year so don't worry
enum pulses{
  Mung = 0;
  ToovarDaal = 1;
  Rajma = 2;
  Urad = 3;
  Vatana = 4;
}

enum status {
  Completed = 0;
  OrderConfirmed = 1;
  TransportDecided = 2;
  Invalid = 3;
  OrderAdded = 4;
}


message assets
{
  required uint32 weight =1 ;
  repeated string previous_owner_pubkey = 2;
  repeated uint32 previous_owner_pincode = 3;
  required string current_owner_pubkey = 4;
  required uint32 current_owner_pincode = 5;
  required type type_of_food = 6;
  optional pulses Pulses = 7;
  optional grains Grains = 8;
  optional fruits_long Fruits_long = 9;
  optional fruits_short Fruits_short = 10;
  optional vegetable_short Vegetable_short = 11;
  optional vegetable_long  Vegetable_long=12 ;
  optional string transporter_pubkey =13;
  required uint64 timestamp = 14;
  repeated string previous_asset_pubkey = 15;
  required status Status = 16;
  required string public_key = 17;
}
